### PR TITLE
Fix compat.rs for `cfg(miri)`

### LINKED ISSUE #99982: `miri` no longer builds after rust-lang/rust#99553
Hello, this is your friendly neighborhood mergebot.
After merging PR rust-lang/rust#99553, I observed that the tool miri has failing tests.
A follow-up PR to the repository https://github.com/rust-lang/miri is needed to fix the fallout.

cc @ChrisDenton, do you think you would have time to do the follow-up work?
If so, that would be great!

### PR DESCRIPTION
**Pull request by ChrisDenton**
Fixes #99982

### CODE REVIEW

**Comment by ChrisDenton** [comment]
r? libs

File: library/std/src/sys/windows/compat.rs [hunk 1]
```diff
@@ -262,7 +262,7 @@ macro_rules! compat_fn_optional {
             #[allow(unused)]
             pub(in crate::sys) fn preload(module: Module) {
                 unsafe {
-                    let symbol_name = ansi_str!(sym $symbol);
+                    static symbol_name: &CStr = ansi_str!(sym $symbol);
```
Comment (RalfJung) [inline library/std/src/sys/windows/compat.rs:265]: Shouldn't statics be named in UPPER_CASE?

Comment (ChrisDenton) [reply library/std/src/sys/windows/compat.rs:265]: Oh hm. And the checker can't see in macros. One sec...

**Comment by ChrisDenton** [comment]
Just uppercased statics and squished.

@bors r=Mark-Simulacrum

**Pull request merged**